fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/EliminateGroupBy.java

```diff
@@ -50,8 +53,11 @@ public Rule build() {
                     Set<Slot> groupby = agg.getGroupByExpressions().stream().map(e -> (Slot) e)
                             .collect(Collectors.toSet());
                     Plan child = agg.child();
-                    boolean unique = child.getLogicalProperties().getFunctionalDependencies()
-                            .isUniqueAndNotNull(groupby);
+                    //boolean unique = child.getLogicalProperties().getFunctionalDependencies()
+                    //        .isUniqueAndNotNull(groupby);
+                    ImmutableSet<FdItem> fdItems = child.getLogicalProperties().getFdItems();
+                    boolean unique = !groupby.isEmpty() && fdItems.stream().anyMatch(e ->
+                            e.isUnique() && !e.isCandidate() && groupby.containsAll(e.getParentExprs()));
```

Comment:
It's incomplete. such as :
```
select * from (select a, b, sum(a + b) as c group by a, b) group by a b c
```
a c are unique, and unique a, b, c also can be infered . That is the uniqueness can be inferred by uniqueness of any subset